pkg/k8s/watchers/watcher.go

```diff
@@ -637,7 +642,11 @@ func (k *K8sWatcher) k8sServiceHandler() {
 				return
 			}
 
-			translator := k8s.NewK8sTranslator(event.ID, *event.Endpoints, true, svc.Labels, true)
+			var oldEP k8s.Endpoints
+			if event.OldEndpoints != nil {
+				oldEP = *event.OldEndpoints
+			}
```

Comment:
There's no notion of "old" in other event types like "delete", there's only the "current" object. Only the "update" event type has "old" and "new", so we don't need to set `OldEndpoints` for any other event type besides `UpdateEndpoints`.

Reply:
I mean, I agree that in case of deletes all the endpoints that are part of the event are the old ones (and it is not necessary to to set `OldEndpoints`). What seems strange to me is that they are passed as `newEPs` to `k8s.NewK8sTranslator()`, and that AFAIU that wil result in `result.PrefixesToRelease` being always empty.

Reply:
Ah now I understand what you're referring to. 

So notice the 4th parameter to `NewK8sTranslator()` -- `revert bool`. This means it will take the "new endpoints" and do the inverse operation, as in it will release them instead of add them.

This means I should simply remove the highlighted code here to avoid confusion and just pass an empty `Endpoints` object so it's clearer. Good catch. I probably just added this snippet of code to be consistent with the `UpdateService` case, but then later realized that the delete path does not need the "old" object and forgot to cleanup this code.

Reply:
Right, I completely missed the revert parameter. Thanks for the clarification.

I still think there's an issue though (due to the subsequent modifications). The only place where revert seems to be used is [here](https://github.com/cilium/cilium/blob/pr%2Fchristarazi%2Fto-services-fixups/pkg/k8s/rule_translate.go#L48-L53), where it guards the population of the egress rules (which is obviously correct). But with the new changes `depopulateEgress` [operates on the `OldEndpoint` object](https://github.com/cilium/cilium/blob/pr%2Fchristarazi%2Fto-services-fixups/pkg/k8s/rule_translate.go#L83). And given that it is always empty nothing will happen. I feel that the revert parameter could be dropped, swapping the parameters in case of deletions:

```go
translator := k8s.NewK8sTranslator(event.ID, *event.Endpoints, k8s.Endpoints{}, svc.Labels)
```

Reply:
Thanks! 

I opted to keep the revert boolean for now and just go for the quickest change, which is that the old and new object are set to the same object with `revert=true`.